### PR TITLE
session_2_kid-116

### PR DESCRIPTION
**Pull request by kid-116**

### CODE REVIEW

File: session_2/student_registry/db/migrate/20210114070046_create_students.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class CreateStudents < ActiveRecord::Migration[6.1]
+  def change
+    create_table :students do |t|
+      t.string :name
+      t.string :roll_number
+      t.string :branch
+      t.decimal :cgpa, precision: 4, scale: 2
```
Comment (abhishekkumar2718) [inline session_2/student_registry/db/migrate/20210114070046_create_students.rb:7]: Can we save a student with cgpa `15.00`? That's wrong, right?
Can you override `cgpa=` function (or write validations) to make it work?

The function should take any floating value, throw `ArgumentError` exception if cgpa > 10 or < 0 and store it appropriately.

https://api.rubyonrails.org/classes/ActiveRecord/Base.html#class-ActiveRecord::Base-label-Overwriting+default+accessors

https://guides.rubyonrails.org/active_record_validations.html

Do try your hand at it! :v:

**Review by abhishekkumar2718 — APPROVED** [review]

**Comment by abhishekkumar2718** [comment]
Nice work on solving Session 2's problems :partying_face:

File: session_2/student_registry/app/models/student.rb [hunk 2]
```diff
@@ -1,2 +1,12 @@
+class CGPAValidator < ActiveModel::Validator
```
Comment (abhishekkumar2718) [inline session_2/student_registry/app/models/student.rb:1]: You don't need to write your own validation here. A simple 
```ruby
validates :cgpa, numericality: {
  greater_than_equal_to: 0.0,
  less_than_equal_to_to: 10.0
}
```

Should work

https://guides.rubyonrails.org/active_record_validations.html#numericality

**Review by abhishekkumar2718 — APPROVED** [review]

File: session_2/student_registry/app/models/student.rb (hunk 2 above)
Comment (kid-116) [reply session_2/student_registry/app/models/student.rb:1]: hmm ok ty

**Pull request closed without merging**